.pipelines/templates/release-githubtasks.yml

```
@@ -26,139 +32,149 @@ jobs:
+      script: |
+        $localInstallerPath = Get-ChildItem -Path "$(Pipeline.Workspace)/GitHubPackages" -Filter '*win-x64.msi' | Select-Object -First 1 -ExpandProperty FullName
+        if (Test-Path -Path $localInstallerPath) {
+          Write-Verbose -Verbose "Installer found at $localInstallerPath"
+        } else {
+          throw "Installer not found"
+        }
+        Write-Verbose -Verbose "Installing PowerShell via msiexec"
+        Start-Process -FilePath msiexec -ArgumentList "/package $localInstallerPath /quiet REGISTER_MANIFEST=1" -Wait -NoNewWindow
+        $pwshPath = Get-ChildItem -Directory -Path 'C:\Program Files\PowerShell\7*' | Select-Object -First 1 -ExpandProperty FullName
+        if (Test-Path -Path $pwshPath) {
+          Write-Verbose -Verbose "PowerShell installed at $pwshPath"
+          Write-Verbose -Verbose "Adding pwsh to env:PATH"
+          Write-Host "##vso[task.prependpath]$pwshPath"
+        } else {
+          throw "PowerShell not installed"
+        }
+    displayName: Install pwsh 7
 
-  - task: AzurePowerShell@5
-    displayName: Download packages from Azure Storage
+  - task: PowerShell@2
     inputs:
-      azureSubscription: az-blob-cicd-infra
-      scriptType: inlineScript
-      azurePowerShellVersion: LatestVersion
+      targetType: inline
       pwsh: true
-      inline: |
-        $storageAccount = "$(StorageAccount)"
-        $containerName = "$(AzureVersion)"
-        $destinationPath = "$(System.ArtifactsDirectory)"
-
-        # Get storage account context
-        $storageContext = New-AzStorageContext -StorageAccountName $storageAccount
-
-        $blobList = Get-AzStorageBlob -Container $containerName -Context $storageContext
-        foreach ($blob in $blobList) {
-          $blobName = $blob.Name
-          $destinationFile = Join-Path -Path $destinationPath -ChildPath $blobName
-          Get-AzStorageBlobContent -Container $containerName -Blob $blobName -Destination $destinationFile -Context $storageContext -Force
-          Write-Output "Downloaded $blobName to $destinationFile"
+      script: |
+        Write-Verbose -Verbose "Pwsh 7 Installed"
+        Write-Verbose -Verbose "env:Path: "
+        $env:PATH -split ';' | ForEach-Object {
+          Write-Verbose -Verbose $_
         }
+    displayName: Check pwsh 7 installation
 
-        $packagesPath = Get-ChildItem -Path $destinationPath\*.deb -Recurse -File | Select-Object -First 1 -ExpandProperty DirectoryName
-        Write-Host "sending -- vso[task.setvariable variable=PackagesRoot]$packagesPath"
-        Write-Host "##vso[task.setvariable variable=PackagesRoot]$packagesPath"
+  - task: PowerShell@2
+    inputs:
+      targetType: inline
+      pwsh: true
+      script: |
+        Write-Verbose -Verbose "Deploy Box Product Pathway Does Not Support the `"checkout`" task"
+        git clone https://$(mscodehubCodeReadPat)@mscodehub.visualstudio.com/PowerShellCore/_git/PowerShell '$(Pipeline.Workspace)/PowerShell'
```

```suggestion
        git clone --branch $(Build.SourceBranch)  --depth  https://$(mscodehubCodeReadPat)@mscodehub.visualstudio.com/PowerShellCore/_git/PowerShell '$(Pipeline.Workspace)/PowerShell'
```
I don't think we use the history here either

---

I update my recommendation to add the checkout to the clone, which will also speed up the clone.

---

Can we make the changelog a build job with an artifact output and update this so we don't need the powershell repo?

---

We can make the CL and Tools clone an artifact output in a separate stage.